Serf Update

Kaytila opened the pull request:
ContinuousGFXUISelector.java
added some checks for right clicks to make the focus loosing go away.

Util.java
the random number generation was weird.

The rest is formatting, I unfortunately have a very different way of formatting.

Commit:
1

Commit:
e

slashman commented on .classpath:
```diff
@@ -13,5 +13,6 @@
 	<classpathentry kind="lib" path="lib/basicPlayer/vorbisspi1.0.2.jar"/>
 	<classpathentry kind="lib" path="lib/basicPlayer/basicplayer3.0.jar"/>
 	<classpathentry combineaccessrules="false" kind="src" path="/libjcsi"/>
+	<classpathentry kind="lib" path="C:/Users/Claus/Downloads/log4j-1.2.17/apache-log4j-1.2.17/log4j-1.2.17.jar"/>
```
in case this is needed, the jar file should be added to the lib directory; we might also want to look forward to change to maven or gradle. Or may be we are just fine.

Kaytila replied:
Good point, will clean that up for the next time.

slashman commented on src/net/slashie/utils/Util.java:
```diff
@@ -1,113 +1,132 @@
-		//Debug.say("low " +low+" hi "+hi+"="+ret);
+		// return (int)((Math.random() * (hi -low))+low);
+		// rand.setSeed(rand.nextLong());
+		int ret = (int) ((rand.nextDouble() * (hi - low + 1)) + low);
```
I don't see any change here? can you point to the actual change on Util.java?

Kaytila replied:
Good point, my notes were wrong. I reverted and its just formatting.

slashman commented on src/net/slashie/serf/ui/oryxUI/ContinuousGFXUISelector.java:
```diff
@@ -96,195 +88,263 @@ public void keyPressed(KeyEvent e) {
+		{
+			public void mousePressed(final MouseEvent e)
+			{
+				if (SwingUtilities.isRightMouseButton(e))
```
Please confirm if the SwingUtilities.isRightMouseButton checks are the ones you added?

Kaytila replied:
yes, to get rid of the loose focus issue

slashman commented on src/net/slashie/serf/ui/oryxUI/GFXUserInterface.java:
```diff
@@ -1161,6 +1166,9 @@ public void shutdown(){
 		si.cleanLayer(getItemsLayer());
 		si.commitLayer(getItemsLayer(), true);
 		
+		si.getComponentsPanel().removeAll();
```
Why was this needed? Memory management?

Kaytila replied:
no, not memory, the Black Frame stayed otherwise, even with the game being shut down.

Commit:
Commit of the day for Serf:

some general split() additions.

I always call trim() after split().

Also added a few LayoutPositions for additional UI elements.
Removed some logger elements.

Changed reduce in Equipment: 
If the quantity of an equipment is smaller than the amount to be
removed, just set to 0 instead of going negative. Do not see where this
might make sense.

slashman commented:
@Kaytila I squashed these 3 commits into a new one, in order to keep them descriptive, please rebase
